UnityProject/Assets/Scripts/Weapons/Projectiles/Behaviours/KeneticProjectileDamageInterity.cs

```diff
@@ -0,0 +1,55 @@
+﻿using UnityEngine;
+using ScriptableObjects.Gun;
+
+namespace Weapons.Projectiles.Behaviours
+{
+	/// <summary>
+	/// Damages integrity on collision
+	/// only for kenetic weapons
+	/// </summary>
+	public class KeneticProjectileDamageInterity : MonoBehaviour, IOnShoot, IOnHit
```

Comment:
You have two typos here. `Kenetic` and `Interity`. I would rename it to `ProjectileDamageIntegrityKinetic`.